webapp/channels/src/components/user_settings/display/user_settings_section.tsx

```diff
@@ -0,0 +1,320 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+/* eslint-disable max-lines */
+
+import React from 'react';
+
+import {FormattedMessage} from 'react-intl';
+import {PrimitiveType, FormatXMLElementFn} from 'intl-messageformat';
+
+import SettingItem from 'components/setting_item';
+import SettingItemMax from 'components/setting_item_max';
+
+type ChildOption = {
+    id: string;
+    message: string;
+    value: string;
+    display: string;
+    moreId: string;
+    moreMessage: string;
+};
+
+type Option = {
+    value: string;
+    radionButtonText: {
+        id: string;
+        message: string;
```

Comment:
I have feelings against this way to pass texts. If ask the caller to send the text directly (created with `intl.formatMessage`) we won't need the `t` function, and I feel everything is simpler.

Also, from what I read from the documentation (https://formatjs.io/docs/react-intl/components/#formattedmessage), the `FormattedMessage` component has no extra benefits, besides maybe limit what elements get re-rendered in case of changes in the language, but that is not often enough to sacrifice other niceties like not using the `t` function and readability.